src/m_gsm_symbolic/sofie_test_parser.py

```diff
@@ -0,0 +1,225 @@
+import json
+import random
+import re
+from dataclasses import dataclass
+from typing import Self
+
+
+functions = {
+    "range": lambda start, stop, step=1: range(start, stop, step), # step default 1, so it can handle 2 and 3 arguments
+    "sample": lambda items, n = 2: random.sample(items, n), # trying to make the dict handle multiple items 
+    "divides": lambda a, b: a % b == 0,}
+
+
+def eval_braces(match, num_values):
+    """ Helper function to evalulate the expressions inside the {}"""
+    expr = match.group(1)  # Get the expression inside the {}
+    try:
+        result = eval(expr, {}, num_values)  # Evaluate the math
+        return str(result)  # Replace with the result
+    except Exception as e:
+        return match.group(0)
+
+
+def eval_expressions(text: str, new_values: dict[str, str]) -> str:
+    '''
+    Evaluate all expressions like {age1+age_diff} and <<expression=result>> in the text.
+    Only keep numeric values. Uses the eval_braces helper function to replace expressions inside brace
+    '''
+    num_values = {}
+    for i, v in new_values.items():
+        try:
+            num_values[i] = int(v) # Try converting the value to an integer
+        except ValueError:
+            pass  # Skip non-numeric values like names    
+    text = re.sub(r"\{([^,{}][^{}]*)\}", lambda match: eval_braces(match, num_values), text)
+    return text
+
+
+@dataclass
+class GeneratedQuestion:
+    question: str
+    answer: str
+    id_orig: int
+    id_shuffled: int
+
+
+@dataclass
+class AnnotatedQuestion:
+    question: str
+    answer: str
+    id_orig: int
+    id_shuffled: int
+    question_annotated: str
+    answer_annotated: str
```

Comment:
```suggestion
@dataclass
class Question:
    question: str
    answer: str
    id_orig: int
    id_shuffled: int


@dataclass
class AnnotatedQuestion(Question):
    question_annotated: str
    answer_annotated: str
```

AnnotatedQuestion is also a Question.